libpod/image/pull.go

```diff
@@ -113,6 +113,52 @@ func (ir *Runtime) getSinglePullRefPairGoal(srcRef types.ImageReference, destNam
 	return singlePullRefPairGoal(rp), nil
 }
 
+// pullGoalFromTarManifest creates a pull goal based on the specified
+// ManifestItem slice. If present, the goal will be generated based on the
+// first item in the slice.
+func (ir *Runtime) pullGoalFromTarManifest(ctx context.Context, manifest *tarfile.ManifestItem, srcRef types.ImageReference, sc *types.SystemContext) (*pullGoal, error) {
+	if manifest == nil {
+		// use the hex of the digest if no manifest is found
+		reference, err := getImageDigest(ctx, srcRef, sc)
+		if err != nil {
+			return nil, err
+		}
+		return ir.getSinglePullRefPairGoal(srcRef, reference)
+	}
+
+	if len(manifest.RepoTags) == 0 {
+		// If the input image has no repotags, we need to feed it a dest anyways
+		digest, err := getImageDigest(ctx, srcRef, sc)
+		if err != nil {
+			return nil, err
+		}
+		return ir.getSinglePullRefPairGoal(srcRef, digest)
+	}
+
+	// Need to load in all the repo tags from the manifest
+	res := []pullRefPair{}
+	for _, dst := range manifest.RepoTags {
+		// Check if image exists and give a warning of untagging
+		localImage, err := ir.NewFromLocal(dst)
+		imageID := strings.TrimSuffix(manifest.Config, ".json")
```

Comment:
(I know this is pre-existing.)

https://github.com/moby/moby/blob/749d90e10f989802638ae542daf54257f3bf71f2/image/spec/v1.2.md#combined-image-json--filesystem-changeset-format does not say the file name of `manifest.Config` is a digest; this should read the manifest and use `ConfigInfo`, or maybe exactly `c/image/Manifest.ImageID` like c/image/storage does.